src/components/VmDetails/cards/DisksCard/DiskImageEditor.js

```diff
@@ -0,0 +1,428 @@
+    this.changesMade = true
+  }
+
+  render () {
+    const { idPrefix, disk, trigger } = this.props
+
+    const createMode = !disk
+    const isImage = disk && disk.get('type') === 'image'
+    const isDirectLUN = disk && disk.get('type') === 'lun'
+
+    const diskSize = disk && (disk.get('lunSize') ? disk.get('lunSize') : disk.get('provisionedSize'))
+
+    return <React.Fragment>
+      {React.cloneElement(trigger, { onClick: this.open })}
+
+      <Modal
+        dialogClassName={style['editor-modal']}
+        id={`${idPrefix}-modal`}
+        show={this.state.showModal}
+        onHide={this.close}
+      >
+        <Modal.Header>
+          <Modal.CloseButton id={`${idPrefix}-modal-close`} onClick={this.close} />
+          <Modal.Title>{createMode ? msg.createNewDisk() : msg.editDisk()}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+
+          <Form horizontal>
+            {/* Alias */}
+            <FormGroup controlId={`${idPrefix}-alias`} className={createMode && 'required'}>
+              <LabelCol sm={3}>
+                { msg.diskEditorAliasLabel() }
+              </LabelCol>
+              <Col sm={9}>
+                <FormControl
+                  type='text'
+                  defaultValue={this.state.alias}
+                  onChange={this.changeAlias}
+                />
+              </Col>
+            </FormGroup>
+
+            {/* Size Display (for edit mode) */}
+            { !createMode &&
+            <FormGroup controlId={`${idPrefix}-size-info`}>
+              <LabelCol sm={3}>
+                { msg.diskEditorSizeLabel() }
+                { !isImage &&
+                  <FieldLevelHelp
+                    inline
+                    content={msg.diskEditorSizeCantChangeHelp()}
+                    buttonClass={style['editor-field-help']}
+                  />
+                }
+              </LabelCol>
+              <Col sm={9}>
+                <div id={`${idPrefix}-size-info`} className={style['editor-field-read-only']}>
+                  { diskSize / 1024 ** 3 }
+                </div>
+              </Col>
+            </FormGroup>
+            }
+
+            {/* Size Editor (initial size for create, expand by size for edit) */}
+            { (createMode || isImage) &&
+            <FormGroup controlId={`${idPrefix}-size`} className={createMode && 'required'}>
+              <LabelCol sm={3}>
+                { createMode &&
+                  <React.Fragment>
+                    {msg.diskEditorSizeLabel()}
+                    <FieldLevelHelp
+                      inline
+                      content={msg.diskEditorSizeCreateHelp()}
+                      buttonClass={style['editor-field-help']}
+                    />
+                  </React.Fragment>
+                }
+                { !createMode && msg.diskEditorResizeLabel() }
+              </LabelCol>
+              <Col sm={9}>
+                { (createMode || isImage) &&
+                  <FormControl
```

Comment:
I still not see here id.

Reply:
And you won't, but it is there:
![screenshot from 2018-11-02 08-59-05](https://user-images.githubusercontent.com/3985964/47916787-be501e00-de7d-11e8-8a04-5f631b990922.png)

When you use a `FormGroup` with a `FormControl`, the id is specified on the `FormGroup` instead: https://react-bootstrap.github.io/components/forms/#forms-props-form-group